### DISCUSSION TITLE
Any idea how to implement infinitely deep stacks?

### DISCUSSION THREAD

**Discussion by CodySwannGT** [expo/router#496] [Q&A]
Here's an example.

1. I'm on Twitter's home page
2. I click to expand a tweet, which pushes the tweet detail screen onto the stack
3. In the tweet detail, I click the author's profile pic, which pushes user detail screen onto the stack and displays a list of the author's tweets
4. I click one of the author's tweets, which pushes another tweet detail screen onto the stack and it displays replies to the tweet
5. I click one of the the replies' author's profile pic, which pushes another user detail screen onto the stack
6. and on and on and on
7. Then as I click "close" or "back" I am taken one screen back at a time until I'm back at the top.

[Here's a video to illustrate how we're currently doing it with react navigation](https://www.dropbox.com/s/m8zviyutuyjt746/Screen%20Recording%202023-04-14%20at%2010.25.05%20AM.mov?dl=0)

**Comment by wcastand** [comment]
my first thought would be to use dynamic route name to generate each screen as an unique id and prevent `router.push` to go back and always add to the stack.

so you would get something like:
`app/[tweet-details].tsx`
`app/[tweet-author].tsx`
`app/[tweet-list].tsx`

...

you can't use the tweet id for the dynamic route otherwise coming back to the same tweet would make router.push go back in the stack instead of adding into it.
